crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1043,21 +1043,31 @@ impl<TX: DbTx> BlockReader for DatabaseProvider<TX> {
     /// will return None.
     fn block_with_senders(
         &self,
-        block_number: BlockNumber,
+        block_number: BlockHashOrNumber,
```

Comment:
not a block number anymore
```suggestion
        block: BlockHashOrNumber,
```